File: PathoFroh/jpametamodel/com/patho/main/model/patient/Slide_.java
package com.patho.main.model.patient;

import com.patho.main.model.StainingPrototype;
import javax.annotation.Generated;
import javax.persistence.metamodel.SingularAttribute;
import javax.persistence.metamodel.StaticMetamodel;

@Generated(value = "org.hibernate.jpamodelgen.JPAMetaModelEntityProcessor")
@StaticMetamodel(Slide.class)
public abstract class Slide_ {

	public static volatile SingularAttribute<Slide, Block> parent;
	public static volatile SingularAttribute<Slide, String> slideID;
	public static volatile SingularAttribute<Slide, StainingPrototype> slidePrototype;
	public static volatile SingularAttribute<Slide, Boolean> idManuallyAltered;
	public static volatile SingularAttribute<Slide, Boolean> reStaining;
	public static volatile SingularAttribute<Slide, Long> completionDate;
	public static volatile SingularAttribute<Slide, Long> id;
	public static volatile SingularAttribute<Slide, Integer> uniqueIDinTask;
	public static volatile SingularAttribute<Slide, Long> creationDate;
	public static volatile SingularAttribute<Slide, Long> version;
	public static volatile SingularAttribute<Slide, Boolean> stainingCompleted;
	public static volatile SingularAttribute<Slide, String> commentary;

}

File: PathoFroh/src/main/kotlin/com/patho/main/model/patient/Slide.kt
package com.patho.main.model.patient

import com.patho.main.model.AbstractPersistable
import com.patho.main.model.StainingPrototype
import com.patho.main.model.interfaces.ID
import com.patho.main.model.interfaces.IdManuallyAltered
import com.patho.main.model.interfaces.Parent
import org.hibernate.annotations.DynamicUpdate
import org.hibernate.annotations.SelectBeforeUpdate
import org.hibernate.envers.Audited
import org.hibernate.envers.NotAudited
import javax.persistence.*

@Entity
@Audited
@SelectBeforeUpdate(true)
@DynamicUpdate(true)
@SequenceGenerator(name = "slide_sequencegenerator", sequenceName = "slide_sequence")
open class Slide : AbstractPersistable, Parent<Block>, IdManuallyAltered {

    @Id
    @GeneratedValue(generator = "slide_sequencegenerator")
    @Column(unique = true, nullable = false)
    open override var id: Long = 0

    @Version
    open var version: Long = 0

    @Column
    open var uniqueIDinTask: Int = 0

    @Column
    open var slideID = ""

    @Column
    open override var idManuallyAltered: Boolean = false

    @Column
    open var creationDate: Long = 0

    @Column
    open var completionDate: Long = 0

    @Column
    open var stainingCompleted: Boolean = false

    @Column
    open var reStaining: Boolean = false

    @Column(columnDefinition = "VARCHAR")
    open var commentary = ""

    @Column(columnDefinition = "VARCHAR")
    open var slideLabelText = ""

    @OneToOne
    @NotAudited
    open var slidePrototype: StainingPrototype? = null

    @ManyToOne
    open override var parent: Block? = null

    constructor() {}

    /**
     * Returns task
     */
    open override val task: Task?
        @Transient
        get() = parent?.task


    /**
     * Return parent
     */
    open override val patient: Patient?
        @Transient
        get() = parent?.patient

}